File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBatteryInfoModel.kt
//
//  IDOBatteryInfoModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.FieldNamingPolicy
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get battery information event number

open class IDOBatteryInfoModel(
    type: Int,
    voltage: Int,
    status: Int,
    level: Int,
    lastChargingYear: Int,
    lastChargingMonth: Int,
    lastChargingDay: Int,
    lastChargingHour: Int,
    lastChargingMinute: Int,
    lastChargingSecond: Int,
    mode: Int
) : IDOBaseModel {

    /// Battery type: 0: Rechargeable lithium battery, 1: Button battery
    @SerializedName("type")
    var type: Int = type

    /// Voltage
    @SerializedName("voltage")
    var voltage: Int = voltage

    /// Battery status
    /// 0: Normal
    /// 1: Charging
    /// 2: Charging complete
    /// 3: Low battery
    @SerializedName("status")
    var status: Int = status

    /// Level
    @SerializedName("level")
    var level: Int = level

    /// Last charging time, year
    @SerializedName("last_charging_year")
    var lastChargingYear: Int = lastChargingYear

    /// Last charging time, month
    @SerializedName("last_charging_month")
    var lastChargingMonth: Int = lastChargingMonth

    /// Last charging time, day
    @SerializedName("last_charging_day")
    var lastChargingDay: Int = lastChargingDay

    /// Last charging time, hour
    @SerializedName("last_charging_hour")
    var lastChargingHour: Int = lastChargingHour

    /// Last charging time, minute
    @SerializedName("last_charging_minute")
    var lastChargingMinute: Int = lastChargingMinute

    /// Last charging time, second
    @SerializedName("last_charging_second")
    var lastChargingSecond: Int = lastChargingSecond

    /// 0: Invalid
    /// 1: Normal mode (non-power saving mode)
    /// 2: Power saving mode
    @SerializedName("mode")
    var mode: Int = mode




    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBatteryInfoModel.swift
//
//  IDOBatteryInfoModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get battery information event number
@objcMembers
public class IDOBatteryInfoModel: NSObject, IDOBaseModel {
    /// Battery type: 0: Rechargeable lithium battery, 1: Button battery
    public var type: Int
    /// Voltage
    public var voltage: Int
    /// Battery status
    /// 0: Normal
    /// 1: Charging
    /// 2: Charging complete
    /// 3: Low battery
    public var status: Int
    /// Level
    public var level: Int
    /// Last charging time, year
    public var lastChargingYear: Int
    /// Last charging time, month
    public var lastChargingMonth: Int
    /// Last charging time, day
    public var lastChargingDay: Int
    /// Last charging time, hour
    public var lastChargingHour: Int
    /// Last charging time, minute
    public var lastChargingMinute: Int
    /// Last charging time, second
    public var lastChargingSecond: Int
    /// 0: Invalid
    /// 1: Normal mode (non-power saving mode)
    /// 2: Power saving mode
    public var mode: Int
    
    enum CodingKeys: String, CodingKey {
        case type = "type"
        case voltage = "voltage"
        case status = "status"
        case level = "level"
        case lastChargingYear = "last_charging_year"
        case lastChargingMonth = "last_charging_month"
        case lastChargingDay = "last_charging_day"
        case lastChargingHour = "last_charging_hour"
        case lastChargingMinute = "last_charging_minute"
        case lastChargingSecond = "last_charging_second"
        case mode = "mode"
    }
    
    public init(type: Int,voltage: Int,status: Int,level: Int,lastChargingYear: Int,lastChargingMonth: Int,lastChargingDay: Int,lastChargingHour: Int,lastChargingMinute: Int,lastChargingSecond: Int,mode: Int) {
        self.type = type
        self.voltage = voltage
        self.status = status
        self.level = level
        self.lastChargingYear = lastChargingYear
        self.lastChargingMonth = lastChargingMonth
        self.lastChargingDay = lastChargingDay
        self.lastChargingHour = lastChargingHour
        self.lastChargingMinute = lastChargingMinute
        self.lastChargingSecond = lastChargingSecond
        self.mode = mode
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

